### PR TITLE
server(sessions): validate configured session file path to prevent pa…

### LINKED ISSUE #1569: [Security] Path traversal vulnerability in FileSystemSessionStore
## Description
The `FileSystemSessionStore` class does not validate the `filePath` configuration parameter, allowing potential path traversal attacks where an attacker could write session data to arbitrary filesystem locations.

## Context
- File: `libraries/typescript/packages/mcp-use/src/server/sessions/stores/filesystem.ts:40-45`
- Component: Session storage backend

## Current Behavior
The `filePath` parameter in `FileSystemSessionStoreConfig` is passed directly to Node.js filesystem operations without validation:

```typescript
constructor(config: FileSystemSessionStoreConfig = {}) {
  this.filePath =
    config.path ?? join(process.cwd(), ".mcp-use", "sessions.json");
  // ...
}
```

If a malicious actor can control `config.path` (via environment variables, config files, or API inputs), they can use `../../../../../etc/passwd` or similar patterns to write to arbitrary locations.

## Expected Behavior
The `filePath` should be validated to ensure it:
1. Is an absolute path or relative to a trusted directory
2. Does not contain path traversal sequences (`../`, `..\\`)
3. Is within allowed boundaries

## Steps to Reproduce
```typescript
import { FileSystemSessionStore } from 'mcp-use/server';

// Malicious config
const store = new FileSystemSessionStore({
  path: '../../../../../../etc/malicious-sessions.json'
});

await store.set('session-123', { /* session data */ });
// Writes to /etc/malicious-sessions.json
```

## Suggested Fix
```typescript
import { resolve, normalize, isAbsolute } from 'node:path';

private validatePath(path: string): string {
  const resolved = isAbsolute(path)
    ? resolve(path)
    : resolve(process.cwd(), path);

  const normalized = normalize(resolved);

  // Prevent path traversal
  if (!normalized.startsWith(resolve(process.cwd())) && !isAbsolute(path)) {
    throw new Error(
      'FileSystemSessionStore: path must be absolute or relative to cwd'
    );
  }

  return normalized;
}

constructor(config: FileSystemSessionStoreConfig = {}) {
  const rawPath = config.path ?? '.mcp-use/sessions.json';
  this.filePath = this.validatePath(rawPath);
  // ...
}
```

## Impact
- Who is affected: All users of FileSystemSessionStore in production or development
- Severity: medium - Requires control over config, but could lead to:
  - Overwriting system files
  - Information disclosure via crafted file locations
  - Potential privilege escalation if combined with other vulnerabilities

_Positively — happy to submit a PR if this is welcome._

### PR DESCRIPTION
**Pull request by samadrehman**
Changes

Add path validation to the FileSystem session store to prevent path traversal when consumers configure a custom session file path.
Add unit tests that reproduce path-traversal attempts and assert safe behavior.

Implementation Details

(1) . Added a private validatePath(pathStr: string): string to [filesystem.ts](vscode-file://vscode-app/d:/Microsoft%20VS%20Code/f6cfa2ea24/resources/app/out/vs/code/electron-browser/workbench/workbench.html) 
which:
Resolves relative paths against process.cwd(), normalizes them, and verifies the resolved path is contained inside the repository root.
Throws an error when a configured path resolves outside the project directory.

(2) Updated the FileSystemSessionStore constructor to call validatePath for the configured path. 

(3) Added unit test: libraries/typescript/packages/mcp-use/tests/security/filesystem-path-traversal.test.ts to:
Verify traversal attempts are rejected.
Verify valid relative and absolute in-repo paths are accepted and persisted.

(4) Architectural decision: keep validation simple and synchronous during construction so misconfiguration fails fast. Recommended follow-up (non-blocking): switch to path.relative() containment check to avoid subtle Windows/drive-letter edge cases.

TypeScript Checklist

Packages Modified

 docs
 tests
 cli
 create-mcp-use-app
 mcp-use (server)
 mcp-use (client)
 inspector

Pre-commit Checklist

 Ran pnpm lint:fix
 Ran pnpm format
 Ran pnpm build
 Ran pnpm changeset (if user-facing)
 Added unit tests
 Updated docs (none required for this small security fix)

Example Usage (Before)
new FileSystemSessionStore({ path: '../secrets/sessions.json' }) // could escape repo root

Example Usage (After)
new FileSystemSessionStore({ path: '.mcp-use/sessions.json' }) // allowed
new FileSystemSessionStore({ path: '/absolute/path/inside/repo/.mcp-use/sessions.json' }) // allowed
new FileSystemSessionStore({ path: '../outside/sessions.json' }) // throws at construction.

Documentation Updates
None required; behavior is a tightened validation of existing configuration. If desired, add a short note to the server/session-store docs describing valid path constraints.

Testing
Unit tests added: tests/security/filesystem-path-traversal.test.ts — covers rejection of out-of-repo paths and acceptance of valid paths.
Manual test: constructed FileSystemSessionStore with bad path and observed constructor throw.
Integration tests unrelated to this change (OpenAI, HMR, Windows ESM) may still fail in CI due to environment; they are out-of-scope for this PR.

Backwards Compatibility

Non-breaking: only rejects misconfigured paths that would previously write outside the project. This is a security hardening; users who intentionally configured an out-of-repo path will need to update their config.

Issues 

Close  #1569

### CODE REVIEW

File: libraries/typescript/packages/mcp-use/src/server/sessions/stores/filesystem.ts [hunk 1]
```diff
@@ -72,15 +72,48 @@ export class FileSystemSessionStore implements SessionStore {
...
+    // Perform a boundary-aware containment check using path.relative.
+    // On Windows, compare case-insensitively to avoid false negatives.
+    const isWindows = process.platform === "win32";
+    const baseForCompare = isWindows ? cwdRoot.toLowerCase() : cwdRoot;
+    const targetForCompare = isWindows ? normalized.toLowerCase() : normalized;
+
+    const rel = pathRelative(baseForCompare, targetForCompare);
+
+    // Accept when the relative path is empty (same path) or does not begin with '..'
+    if (rel === "" || (!rel.startsWith("..") && !isAbsolute(rel))) {
```
Comment (khandrew1) [inline libraries/typescript/packages/mcp-use/src/server/sessions/stores/filesystem.ts:108]: This would actually be a regression as it only allows paths within the CWD. There would be use cases in which we would need to mount it at `/tmp`

**Review by khandrew1 — COMMENTED** [review]
Hi @samadrehman! Thanks for the PR. Sorry but upon closer inspection of the issue and this PR, I've come to realize this is a non-issue.

> This file path is developer supplied configuration, not user controlled input. Untrusted data does not pass through this caller. Closing for now. If you need to report other security vulnerabilities please use the Github Security tab.
